### PR TITLE
feat: Add CLI and rewrite all scripts to NodeJS in order to improve Windows support.

### PR DESCRIPTION
**Pull request by pziemkowski**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?

feature

### What is the current behavior?

Currently most of the scripts are written for bash/shell so they cannot work directly in Windows. 
Previous approach was to use WSL to run all commands inside but looks like it's too difficult to configure to most users.

### What is the new behavior?

All the scripts that run on the host machine have been rewritten to NodeJS.
I also added a simple local CLI (based on Oclif) to remove dependency on `make`.

### Does this PR introduce a breaking change?

Yes. Removed `make` rules and introduces a CLI.

### CODE REVIEW

File: packages/internal/cli/test/commands/hello/index.test.ts
Comment (mkleszcz) [inline packages/internal/cli/test/commands/hello/index.test.ts:1]: Please remove

File: packages/internal/cli/test/commands/hello/world.test.ts
Comment (mkleszcz) [inline packages/internal/cli/test/commands/hello/world.test.ts:1]: Please remove

File: packages/webapp-libs/webapp-contentful/scripts/download-graphql-schema.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+const { spawn } = require('child_process');
+const fs = require('fs-extra');
+const dotenv = require('dotenv');
+function runCommand(command, args) {
```
Comment (mkleszcz) [inline packages/webapp-libs/webapp-contentful/scripts/download-graphql-schema.js:4]: We have this code in couple of places. WDYT about extract it to a single place?

**Review by mkleszcz — CHANGES_REQUESTED** [review]

**Comment by mkleszcz** [comment]
@pziemkowski also we need to adjust the README.md file as well as `packages/backend/README.md` because there are a `make` commands

File: packages/webapp-libs/webapp-contentful/scripts/download-graphql-schema.js (hunk 1 above)
Comment (pziemkowski) [reply packages/webapp-libs/webapp-contentful/scripts/download-graphql-schema.js:4]: Yes and no. Those are very simple nodejs packages that do not require dependance on the whole typescript configuration (paths, pnpm workspace, etc) so it's not easy to create a reusable package. I decided it's not worth it.

File: packages/internal/cli/README.md [hunk 2]
```diff
@@ -0,0 +1,1241 @@
+# oclif-hello-world
```
Comment (mkleszcz) [inline packages/internal/cli/README.md:1]: We still have the example copy here ;)

File: packages/internal/cli/src/lib/runCommand.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { SpawnOptionsWithoutStdio, spawn } from 'node:child_process';
+
+export function runCommand(
+  command: string,
+  args: string[],
+  options?: SpawnOptionsWithoutStdio
+) {
+  return new Promise<void>((resolve, reject) => {
+    console.log([command, ...args].join(' '));
```
Comment (mkleszcz) [inline packages/internal/cli/src/lib/runCommand.ts:9]: It is intended to log command with all args?

File: packages/webapp-libs/webapp-api-client/scripts/download-graphql-schema.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  });
+}
+
+(async () => {
+  try {
+    dotenv.config({ path: '../../webapp/.env' });
+
+    const apiUrl = 'http://localhost:5001/api/graphql/';
+
+    fs.remove('./src/graphql/__generated/types.ts');
```
Comment (mkleszcz) [inline packages/webapp-libs/webapp-api-client/scripts/download-graphql-schema.js:25]: I think this should be `fs.removeSync`

File: packages/webapp-libs/webapp-contentful/scripts/download-graphql-schema.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    if (!VITE_CONTENTFUL_TOKEN || VITE_CONTENTFUL_TOKEN === '<CHANGE_ME>') {
+      console.log('VITE_CONTENTFUL_TOKEN has not been set. Skipping...');
+      return;
+    }
+
+    const CONTENTFUL_URL = `https://graphql.contentful.com/content/v1/spaces/${VITE_CONTENTFUL_SPACE}/environments/${VITE_CONTENTFUL_ENV}?access_token=${VITE_CONTENTFUL_TOKEN}`;
+
+    fs.remove('./src/contentful/__generated/types.ts');
```
Comment (mkleszcz) [inline packages/webapp-libs/webapp-contentful/scripts/download-graphql-schema.js:41]: `removeSync`

**Review by mkleszcz — CHANGES_REQUESTED** [review]

File: packages/internal/cli/src/lib/runCommand.ts (hunk 3 above)
Comment (pziemkowski) [reply packages/internal/cli/src/lib/runCommand.ts:9]: Yes, so you can see what is being executed

File: packages/webapp-libs/webapp-api-client/scripts/download-graphql-schema.js (hunk 4 above)
Comment (pziemkowski) [reply packages/webapp-libs/webapp-api-client/scripts/download-graphql-schema.js:25]: Or just a missing `await`, since those return promises

**Review by mkleszcz — APPROVED** [review]

**Pull request merged**